dbt/include/snowflake/macros/adapters.sql

```diff
@@ -171,7 +171,11 @@
 {% endmacro %}
 
 {% macro snowflake__alter_relation_comment(relation, relation_comment) -%}
-  comment on {{ relation.type }} {{ relation }} IS $${{ relation_comment | replace('$', '[$]') }}$$;
+    {%- set relation_type = relation.type -%}
+    {%- if relation_type == 'dynamic_table' -%}
```

Comment:
Minor nits:

To avoid the hard coded string, this could be:
```
{%- if relation.is_dynamic_table -%}
```
And I would move the initial "set relation_type" into an else statement in this if block.

What you have works, I just think this would be more explicit.